### PR TITLE
Default version with scope test

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
org.wso2.am.integration.tests.version.DefaultVersionWithScopesTestCase

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/resources/oas/v3/defaultVersionScopes.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    "corsConfigurationEnabled": false,
+    "accessControlAllowOrigins": [],
+    "accessControlAllowCredentials": false,
+    "accessControlAllowHeaders": [],
+    "accessControlAllowMethods": []
+  },
+  "x-wso2-production-endpoints": "http://localhost:8780/response",
+  "x-wso2-sandbox-endpoints": "http://localhost:8780/response",
+  "x-wso2-basePath": "/defaultversionscope/1.0.0"
+}
```
Comment (HiranyaKavishani) [inline modules/integration/tests-integration/tests-backend/src/test/resources/oas/v3/defaultVersionScopes.json]: Add a new line to EOF

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java [hunk 2]
```diff
@@ -73,189 +64,161 @@
...
-            requestBody = "grant_type=password&username=" + USER_SAM + "&password=sam123&scope=user_scope";
-            accessTokenGenerationResponse = new JSONObject(apiStore.generateUserAccessKey(consumerKey, consumerSecret,
-                                                                                          requestBody, tokenEndpointURL)
-                                                                   .getData());
-            accessToken = accessTokenGenerationResponse.getString("access_token");
-
-            requestHeaders = new HashMap<String, String>();
-            requestHeaders.put("Authorization", "Bearer " + accessToken);
+            Map<String, String> requestHeaders;
+            requestHeaders = new HashMap<>();
```
Comment (HiranyaKavishani) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java]: Can move to 173

File: modules/integration/tests-integration/tests-backend/src/test/resources/oas/v3/defaultVersionScopes.json (hunk 1 above)
Comment (AmaliMatharaarachchi) [reply modules/integration/tests-integration/tests-backend/src/test/resources/oas/v3/defaultVersionScopes.json]: Fixed.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java (hunk 2 above)
Comment (AmaliMatharaarachchi) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java]: Fixed. Thanks.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java [hunk 3]
```diff
@@ -73,189 +64,157 @@
...
-            apiStore.removeApplication(APP_NAME);
-        }
-
-        if (apiPublisher != null) {
-            apiPublisher.deleteAPI(API_NAME, API_VERSION, API_PROVIDER);
+        SubscriptionListDTO subsDTO = restAPIStore.getAllSubscriptionsOfApplication(applicationID);
+        for (SubscriptionDTO subscriptionDTO : subsDTO.getList()) {
+            restAPIStore.removeSubscription(subscriptionDTO.getSubscriptionId());
         }
+        restAPIStore.deleteApplication(applicationID, null);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java]: This method is updated by removing the tenant domain. Lets remove the null parameter.

Comment (AmaliMatharaarachchi) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java]: Done.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java [hunk 4]
```diff
@@ -19,52 +19,43 @@
...
 import org.apache.axis2.AxisFault;
+import org.apache.commons.io.IOUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.json.JSONObject;
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Test;
+import org.testng.annotations.*;
+import org.wso2.am.integration.clients.store.api.v1.dto.*;
```
Comment (Rajith90) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java]: Lets remove the * imports

Comment (AmaliMatharaarachchi) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/version/DefaultVersionWithScopesTestCase.java]: Fixed in https://github.com/wso2/product-apim/pull/7429

**Pull request merged**